Z/Zstd/bundled/patches/qsort_r.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/lib/common/zstd_deps.h b/lib/common/zstd_deps.h
+index 8a9c7cc5..c6cb2aea 100644
+--- a/lib/common/zstd_deps.h
++++ b/lib/common/zstd_deps.h
+@@ -31,7 +31,7 @@
+  */
+ #if defined(__linux) || defined(__linux__) || defined(linux) || defined(__gnu_linux__) || \
+     defined(__CYGWIN__) || defined(__MSYS__)
+-#if !defined(_GNU_SOURCE) && !defined(__ANDROID__) /* NDK doesn't ship qsort_r(). */
++#if !defined(_GNU_SOURCE) && !defined(__ANDROID__) && !defined(__MUSL__) /* NDK and musl don't ship qsort_r(). */
```

Comment:
`__MUSL__` doesn't exist.

Reply:
I define it:
```
elif [[ "${target}" == *musl* ]]; then
    # Define `__MUSL__` for the patch `qsort_r.patch`
    sed -ri "s/^c_args = \[(.*)\]/c_args = [\1, '-D__MUSL__']/" ${MESON_TARGET_TOOLCHAIN}
```

There isn't otherwise a good way to see whether we're building against musl. I assume the correct solution would be to add a configure check for `qsort_r`, but that would be more work.

Reply:
> There isn't otherwise a good way to see whether we're building against musl.

Yeah, that's intentional, [they purposefully don't want to be special-cased](https://wiki.musl-libc.org/faq).  But our musl libc should have `qsort_r`: #4547

Reply:
`qsort_r` was introduced in musl 1.2.3. If we're using 1.2.2 then we don't have `qsort_r`.

Reply:
I literally patched 1.2.2 to add `qsort_r` in #4547.

Reply:
There is a linker error, complaining that the symbol `qsort_r` is missing. At the same time, zstd doesn't need `qsort_r`, it's happy with `qsort` – it uses qsort as fallback on systems where qsort_r is missing.

Reply:
Ah, now I see: https://github.com/JuliaPackaging/Yggdrasil/blob/15db5a48d1e90a33cb79fc52f9c690771d051cc2/0_RootFS/Rootfs/bundled/libs/libc/download_libcs.sh#L15 is only what we use for running executables/loading libraries within the sandbox, but linking happens with https://github.com/JuliaPackaging/Yggdrasil/blob/15db5a48d1e90a33cb79fc52f9c690771d051cc2/0_RootFS/gcc_sources.jl#L285